File: DMOJ/Another Contest/ACC7/CodeReview.py
import sys
import random
input = sys.stdin.readline

n = int(input())
arr = [i for i in range(1, n+1)]

random.seed(1)
def scramble():
    ret = []
    for l in range(n-1, -1, -1):
        ind = random.randint(0, l)
        arr[l], arr[ind] = arr[ind], arr[l]
        ret.append(arr.pop())

    return ret


while True:
    print(*arr, sep=' ')
    sys.stdout.flush()

    ret = int(input())
    if ret == 0:
        break

    arr = scramble()

File: DMOJ/Another Contest/ACC7/CodeReview.cpp
// Another Contest 7 Problem 2 - Code Review

#include <bits/stdc++.h>
using namespace std;

int n;
vector<int> arr;
void scramble() {
    vector<int> ret;
    for (int l = n; l >= 1; l--) {
        int ind = rand() % l;
        swap(arr[l - 1], arr[ind]);
        ret.push_back(arr.back());
        arr.pop_back();
    }
    arr = ret;
}
int main() {
    scanf("%d", &n);
    for (int i = 1; i <= n; i++) arr.push_back(i);
    while (true) {
        scramble();
        for (int i : arr) printf("%d ", i);
        printf("\n");
        fflush(stdout);

        int in;
        scanf("%d", &in);
        if (!in) break;
    }

    return 0;
}